### PR TITLE
backport gulp config from master

### LINKED ISSUE #2932: Unable to load application on production profile
Hi All,

I just create a new application based on an existing yo-rc.json:

~~~json
{
  "generator-jhipster": {
    "baseName": "manager",
    "packageName": "com.xxx.manager",
    "packageFolder": "com/xx/manager",
    "authenticationType": "oauth2",
    "hibernateCache": "ehcache",
    "clusteredHttpSession": "no",
    "websocket": "spring-websocket",
    "databaseType": "sql",
    "devDatabaseType": "mysql",
    "prodDatabaseType": "mysql",
    "searchEngine": "no",
    "useSass": false,
    "buildTool": "gradle",
    "frontendBuilder": "gulp",
    "enableTranslation": true,
    "enableSocialSignIn": false,
    "rememberMeKey": "456fca32e9876be9b48b542343e321a3c5b800ac",
    "testFrameworks": [
      "cucumber",
      "protractor"
    ]
  }
}
~~~

Then, build the project:
`./gradlew -Pprod bootRepackage`

And run the app:
`java -jar vmaxmanager-0.0.1-SNAPSHOT.war`

The app runs fine:
screen-shot-2016-02-18-18-04-14

 But when I run the app with profile prod:
`java -jar vmaxmanager-0.0.1-SNAPSHOT.war --spring.profiles.active=prod`

The app seem to run ok in the console:
screen-shot-2016-02-18-18-06-30

But, when I try to open the page, I got a 404:
screen-shot-2016-02-18-18-07-32

I have the same issue when I deploy to tomcat the `.war.original` file with `-Dspring.profiles.active=prod` flag

### PR DESCRIPTION
**Pull request by atomfrede**
backport gulp config from master to generate production files in 2.x applications

close #2932

### CODE REVIEW

**Comment by jdubois** [comment]
That's a huge work!!
Do you think I can release a new 2.27.1 version once this is merged?

File: app/templates/_package.json [hunk 1]
```diff
@@ -8,62 +8,40 @@
     "src/main/webapp/bower_components"
   ],
   "dependencies": {
-    "bower": "^1.6.3"<% if (testFrameworks.indexOf('protractor') > -1) { %>,
-    "protractor": "^2.5.1"<% } %>
+    "bower": "^1.6.3"
   },
-  "devDependencies": {<% if (frontendBuilder == 'grunt') { %>
-    "grunt": "0.4.5",
```
Comment (deepu105) [inline app/templates/_package.json]: 2.x still supports grunt so we should keep grunt deps here

File: app/templates/_package.json [hunk 2]
```diff
@@ -88,9 +66,8 @@
   "engines": {
     "node": "^0.12 || ^4.2"
   },
-  "scripts": {<% if (frontendBuilder == 'grunt') { %>
-    "test": "grunt test"<% } if (frontendBuilder == 'gulp') { %>
```
Comment (deepu105) [inline app/templates/_package.json]: This should be kept as well

File: app/templates/_package.json (hunk 1 above)
Comment (atomfrede) [inline app/templates/_package.json]: Oh yes you are right, will fix it!

**Comment by deepu105** [comment]
Plz check my comments before merging

**Comment by deepu105** [comment]
@jdubois yed I think its a good idea to do a 2.27.1 from the 2.x
maintenance branch

**Comment by atomfrede** [comment]
Created a test project with grunt too, works fine now.

**Comment by deepu105** [comment]
Travis is failing

File: app/templates/_package.json [hunk 3]
```diff
@@ -48,22 +47,31 @@
...
     "protractor-jasmine2-screenshot-reporter": "0.1.7",<% } %>
     "gulp-rev": "6.0.1",
+    "gulp-sourcemaps": "^1.6.0",
     "gulp-jshint": "1.12.0",
     "gulp-browserify": "0.5.1",<% if (useSass) { %>
     "gulp-sass": "2.1.0",<% } %>
+    "gulp-notify": "2.2.0",
+    "gulp-plumber" : "1.0.1",
+    "gulp-changed" : "1.3.0",
+    "gulp-cached" : "1.1.0",<% } %>
```
Comment (deepu105) [inline app/templates/_package.json]: This shouldn't end here it should be after run sequence

**Comment by atomfrede** [comment]
Anyone an idea why travis still fails? ``error: ENOENT: no such file or directory, open 'build.gradle'``

**Comment by jdubois** [comment]
I don't see it failing now, can you point to the failing build?

On Sat, Feb 20, 2016 at 9:41 PM, Frederik Hahne <notifications@github.com>
wrote:

> Anyone an idea why travis still fails? error: ENOENT: no such file or
> directory, open 'build.gradle'
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/jhipster/generator-jhipster/pull/2943#issuecomment-186678880>
> .
>



-- 
Julien Dubois

Twitter: @juliendubois <http://twitter.com/#!/juliendubois>
Phone: +33 (0)6 25 02 34 18

**Comment by atomfrede** [comment]
It is fixed now. I copied ``utils.js`` from a generated project and not from the generator :innocent:

**Comment by deepu105** [comment]
@jdubois I have merged this, you can do a maintanance release when you like

**Comment by jdubois** [comment]
Thanks! I'll do a maintenance release very soon

**Pull request merged**